enumerable.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+
+module Enumerable
+  def my_each
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_each_with_index
+    if block_given?
+      (0..length - 1).my_each do |i|
+        yield(self[i], i)
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_select
+    if block_given?
+      new_arr = []
+      (0..length - 1).my_each do |i|
+        new_arr.push(self[i]) if yield(self[i])
+      end
+    else
+      puts "You didn't pass in a block!"
+    end
+  end
+
+  def my_all?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
+
+  def my_any?
+    (0..length - 1).my_each do |element|
+      return false unless yield(element)
+    end
+    true
+  end
```

Comment:
- We need to fix this my_each to make this method work as expected.

- my_any should also behave like the original method:
`['ant', 'bear', 'cat'].any? { |word| word.length >= 3 } #=> true`
`['ant', 'bear', 'cat'].any? { |word| word.length >= 4 } #=> true`
`['ant', 'bear', 'cat'].any?(/d/)                        #=> false`
`[nil, true, 99].any?(Integer)                     #=> true`
`[nil, true, 99].any?                              #=> true`
`[].any?                                           #=> false`